packages/taquito-core/src/error/common.ts

```diff
@@ -1,32 +1,54 @@
+import {
+  ParameterValidationError,
+  PermissionDeniedError,
+  UnsupportedAction,
+  ValidationError,
+} from './errors';
+
 /**
  *  @category Error
  *  @description Error that indicates an invalid address being passed or used (both contract and implicit)
  */
-export class InvalidAddressError extends Error {
+export class InvalidAddressError extends ParameterValidationError {
   public name = 'InvalidAddressError';
   constructor(public address: string, errorDetail?: string) {
-    super();
-    const baseMessage = `The address '${address}' is invalid.`;
-    this.message = errorDetail ? `${baseMessage} ${errorDetail}` : baseMessage;
+    super(
+      errorDetail
+        ? `The address '${address}' is invalid. ${errorDetail}`
+        : `The address '${address}' is invalid.`
+    );
   }
 }
 
 /**
  *  @category Error
  *  @description Error that indicates an invalid derivation path being passed or used
  */
-export class InvalidDerivationPathError extends Error {
+export class InvalidDerivationPathError extends ParameterValidationError {
   public name = 'InvalidDerivationPathError';
   constructor(public message: string) {
     super(message);
   }
 }
 
+/**
+ *  @category Error
+ *  @description Error that indicates an invalid derivation type being passed or used
+ */
+export class InvalidDerivationTypeError extends Error {
```

Comment:
did we miss out on extending this one from higher category or it's the one undecided?

Reply:
ahh thanks for the catch thats a rebase miss. since it disappeared